python/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK/EBS_SNAPSHOT_PUBLIC_RESTORABLE_CHECK_test.py

```diff
@@ -0,0 +1,269 @@
+        first_response = {'NextToken': '{}'.format(next_token),
+                          'ResponseMetadata': {'HTTPHeaders': {'content-type': 'text/xml;charset=UTF-8',
+                            'date': 'Fri, 15 Mar 2019 16:58:27 GMT',
+                            'server': 'AmazonEC2',
+                            'transfer-encoding': 'chunked',
+                            'vary': 'Accept-Encoding'},
+                           'HTTPStatusCode': 200,
+                           'RequestId': '9d292dca-1d15-48aa-886f-5821e15691f2',
+                           'RetryAttempts': 0},
+                           'Snapshots': [{'Description': 'hvm-ssd/ubuntu-tsample151007',
+                                          'Encrypted': False,
+                                          'OwnerId': '123456789012',
+                                          'Progress': '100%',
+                                          'SnapshotId': 'snap-9a0a02f7',
+                                          'StartTime': datetime.datetime(2015, 10, 8, 0, 46, 39, tzinfo=tzutc()),
+                                          'State': 'completed',
+                                          'VolumeId': 'vol-503ab6b0',
+                                          'VolumeSize': 8}]
+                         }
+        final_response = {'ResponseMetadata': {'HTTPHeaders': {'content-type': 'text/xml;charset=UTF-8',
+                          'date': 'Fri, 15 Mar 2019 16:55:26 GMT',
+                          'server': 'AmazonEC2',
+                          'transfer-encoding': 'chunked',
+                          'vary': 'Accept-Encoding'},
+                          'HTTPStatusCode': 200,
+                          'RequestId': 'fa2f8132-c8eb-46c3-b242-4example42a',
+                          'RetryAttempts': 0},
+                          'Snapshots': [{'Description': 'Copied for DestinationAmi ami.',
+                                        'Encrypted': False,
+                                        'OwnerId': '123456789012',
+                                        'Progress': '100%',
+                                        'SnapshotId': 'snap-0daeb11514fba831a',
+                                        'StartTime': datetime.datetime(2019, 3, 15, 14, 58, 48, 662000, tzinfo=tzutc()),
+                                        'State': 'completed',
+                                        'VolumeId': 'vol-ffffffff',
+                                        'VolumeSize': 99}]
+                          }
+        if((NextToken is None) and (OwnerIds[0]=='123456789012') and (RestorableByUserIds[0]=='all') and (MaxResults==1000)):
+            return(first_response)
+        elif(NextToken == next_token):
+            return(final_response)
+
+# Checks for scenario wherein non-compliant resources are present and pagination exists
+    def test_all_noncompliant_resources_with_pagination(self):
```

Comment:
Name the test function which is more aligned with the Gherkin Scenario like:
test_Scenario_1_noncompliant_resources_with_pagination

Reply:
Renamed target functions